thunder/transforms/quantization.py

```diff
@@ -294,3 +296,230 @@ def transform_traces_pre_prologue(self, prologue_trace, computation_trace, epilo
+        self, model: thunder.ThunderModule, submodule_name: str, state_dict: dict
+    ) -> dict:
+        if submodule_name not in self.lora_linear_names:
+            return state_dict
+
+        w = state_dict["weight"]
+        in_features, out_features = w.shape[0], w.shape[1]
+        lora_a = torch.nn.Parameter(w.new_empty((self.r, in_features)))
+        lora_b = torch.nn.Parameter(w.new_empty((out_features, self.r)))
+        self.init_lora_linear(lora_a, lora_b)
+
+        state_dict = state_dict.copy()
+        state_dict[f"weight.lora_a"] = lora_a
+        state_dict[f"weight.lora_b"] = lora_b
+
+        return state_dict
+
+    def transform_traces_pre_prologue(self, prologue_trace, computation_trace, epilogue_trace, **kwargs):
+        tm = self.thunder_module
+        from thunder.core.trace import tracectx
+
+        checks = get_checks(prologue_trace)
+
+        prologue_trace = trace_with_replaced_proxy_metadata(prologue_trace, {})
+
+        checks = get_checks(prologue_trace)
+        proglogue_to_compute_outputs = prologue_trace.output[0]
+        output_idxes = {o.name: i for i, o in enumerate(proglogue_to_compute_outputs)}
+
+        lora_linear_proxies: dict[str, str] = {}  # proxy_name -> param_name
+        additional_proxies: dict[str, thunder.Proxy] = {}  # param_name.(absmax|code) -> Proxy
+
+        new_bsyms = []
+        new_compute_inputs = []
+        for n, qs in self.lora_linear_states.items():
+            param = tm.get_parameter(n)
+            n_lora_a = f"{n}.lora_a"
+            n_lora_b = f"{n}.lora_b"
+            n_linear = f"{n}"
+            check, get_param = checks[n]
+            lora_linear_proxies[get_param.output.name] = n
+            # check has args: tensor, shape, device, dtype, requires_grad
+            proxy, _, device, _, requires_grad = check.args
+
+            output_idx = output_idxes.get(get_param.output.name)
+            if output_idx is not None:
+                with tracectx(prologue_trace):
+                    # better way
+                    proxy_lora_a = thunder.TensorProxy(
+                        name=f"{get_param.output.name}_lora_a",
+                        shape=qs["lora_a.shape"],
+                        dtype=thunder.dtypes.to_dtype(qs["lora_a.dtype"]),
+                        device=thunder.devices.to_device(device),
+                        requires_grad=requires_grad,
+                    )
+                    proxy_lora_b = thunder.TensorProxy(
+                        name=f"{get_param.output.name}_lora_b",
+                        shape=qs["lora_b.shape"],
+                        dtype=thunder.dtypes.to_dtype(qs["lora_b.dtype"]),
+                        device=thunder.devices.to_device(device),
+                        requires_grad=requires_grad,
+                    )
+                    new_bsyms.append(get_param.sym.bind(get_param.args[0], n_lora_a, output=proxy_lora_a))
+                    self.check_proxy(
```

Comment:
Obviously, I forgot this with the quantization transform you modeled this after, but instead of checking the proxy here (which should be always good because we just defined it), let's add checks to the prologue

https://github.com/Lightning-AI/lightning-thunder/commit/8d3561aae877f277bb820ce7cd74057399ccc7ac